### DISCUSSION TITLE
Webp near-lossless

### DISCUSSION THREAD

**Discussion by Worfhard** [ImageMagick/ImageMagick#3029] [Help] [answered]
Hello, I'm wondering if ImageMagick supports the near-lossless encoding option for webp?
I'm trying to use it but without any luck.

**Comment by snibgo** [comment]
WebP encoding options are listed at https://imagemagick.org/script/webp.php

If you have a specific problem, please give more details: your exact command, your inputs and outputs, your IM version and platform, etc.

**Reply by Worfhard** [reply]
I'm using the command `magick test.png -quality 90 -define webp:near-lossless=95 test.webp`
The output file comes out exactly the same as if I would be using `magick test.png -quality 90 -define webp:lossless=false test.webp`

Version: ImageMagick 7.0.10-53 Q16 x64 2020-12-24

Windows 10

**Comment by dlemstra** [comment]
`webp:near-lossless=95` is not a define. The default for lossless is `false` so your command would translate to this in both situations:
`magick test.png -quality 90 test.webp`

**Reply by Worfhard** [reply]
Oh that explains a lot.
So how would I go about using the `near-lossless` compression option? I'm having a hard time figuring it out.

**Comment by snibgo** [comment, marked as answer]
See the web page I linked to. IM has no "near-lossless" option.

**Reply by dlemstra** [reply]
The next version of ImageMagick will have support for this option.
